### PR TITLE
Prevent creation of another Kubevirt

### PR DESCRIPTION
**Pull request by xpivarc**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-operator/webhooks/kubevirt-create-admitter.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package webhooks
```
Comment (fossedihelm) [inline pkg/virt-operator/webhooks/kubevirt-create-admitter.go:1]: missing header

**Review by fossedihelm — COMMENTED** [review]
@xpivarc there is a compute tests that is failing due to this change.
In general, I have some little concerns regarding this webhook, since it is created by kubevirt itself. This means that the first time KV is created, the validation is not performed.
Even if the scope of this is to prevent the "other" kubevirt creation, the presence of the `kubevirt-create-admitter` could lead to think that this can be used for general KV validation; but this is not true. Any thoughts?

**Comment by vladikr** [comment]
> @xpivarc there is a compute tests that is failing due to this change. In general, I have some little concerns regarding this webhook, since it is created by kubevirt itself. This means that the first time KV is created, the validation is not performed. Even if the scope of this is to prevent the "other" kubevirt creation, the presence of the `kubevirt-create-admitter` could lead to think that this can be used for general KV validation; but this is not true. Any thoughts?

@fossedihelm Are you concerned only about the semantics of the `kubevirt-create-admitter` ?  Can we get away with a brief explanation in the code?

>  This means that the first time KV is created, the validation is not performed.
@fossedihelm do you mean that the validation of the KV itself?

**Comment by fossedihelm** [comment]
> > @xpivarc there is a compute tests that is failing due to this change. In general, I have some little concerns regarding this webhook, since it is created by kubevirt itself. This means that the first time KV is created, the validation is not performed. Even if the scope of this is to prevent the "other" kubevirt creation, the presence of the `kubevirt-create-admitter` could lead to think that this can be used for general KV validation; but this is not true. Any thoughts?
> 
> @fossedihelm Are you concerned only about the semantics of the `kubevirt-create-admitter` ? Can we get away with a brief explanation in the code?

@vladikr Yes, I think that an explanation comment in the code can be enough! Basically, I am worried that in the future this `kubevirt-create` webhook will be used to perform some kubevirt fields validation. 
This is something that cannot be done currently, because it the webhook is installed by the operator, and thus kv is already installed.
It is ok to use it for ducplicate prevention, but not for any other validation.

**Comment by acardace** [comment]
/close
in favor of https://github.com/kubevirt/kubevirt/pull/10511

**Pull request closed without merging**